Add version details.

gillessed opened the pull request:
Add a versionDetails() function which returns an object
containing the latest tag name and the commit distance to the tag.

This is useful information for publishing to almanac.

markelliot commented on readme.md:
```diff
@@ -3,10 +3,14 @@ Git-Version Gradle Plugin
+It behaves exactly as the JGit `git describe` method behaves, except that when the repository is in a dirty 
+state, appends `.dirty` to the version string. 
+
+The second, called `versionDetails()`, return an object containing the specific details of the version string:
```
"returnS"

markelliot commented on readme.md:
```diff
@@ -20,6 +24,12 @@ Set the version of a project by calling:
 
+You can get an object containing more detailed information by calling:
+
+    details = versionDetails()
```
`def details =`

markelliot commented on src/main/groovy/com/palantir/gradle/gitversion/GitVersionPlugin.groovy:
```diff
@@ -42,12 +40,42 @@ class GitVersionPlugin implements Plugin<Project> {
+
+            try {
+                Git git = Git.wrap(new FileRepository(gitDir))
+                String description = git.describe().call();
```
for consistency, this really ought to call the same thing as gitVersion()

gillessed replied:
Don't they both call `git.describe()` already?

markelliot replied:
It'd be good to encapsulate the calls. I've been meaning to clean this up to memoize anyway, can leave as-is and I'll come back to it later.

markelliot commented on src/main/groovy/com/palantir/gradle/gitversion/GitVersionPlugin.groovy:
```diff
@@ -42,12 +40,42 @@ class GitVersionPlugin implements Plugin<Project> {
+                  return new VersionDetails(description, 0);
+                }
+
+                String[] split = description.split('-')
```
I think it'd be cleaner here to use a regular expression then to split things up and sow them back together.

markelliot commented:
Looks reasonable, couple of nits + would like to see some tests.

markelliot commented:
looks good to me, squash?

Commit:
Add version details

Add a versionDetails() function which returns an object
containing the latest tag name and the commit distance to the tag.

This is useful information for publishing to almanac.
